### PR TITLE
Feature implement persistence

### PR DESCRIPTION
**Pull request by yxyx-github**
This is the first part. Only save into a json file implemented. Load implementation will be implemented in the future.

### CODE REVIEW

**Comment by vreoio** [comment]
Shouldn't the dependencies be added to the README?

File: src/test/java/hwr/oop/group4/todo/persistence/PersistedTodoListTest.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+        projectA.addTask(taskB);
+        projectA.addTask(taskC);
+
+        todoList.addProject(projectA);
+        todoList.addLoseTask(taskA);
+
+        var persistedTodoList = new PersistedTodoList(todoList);
+
+        // System.out.println(persistedTodoList.toString());
```
Comment (termitehuegel) [inline src/test/java/hwr/oop/group4/todo/persistence/PersistedTodoListTest.java:41]: Do not comment out code

File: src/test/java/hwr/oop/group4/todo/persistence/FileAdapterTest.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+    @Test
+    void save(@TempDir Path tempDir) {
+        Persisted data = mock();
+        when(data.toString()).thenReturn("demo file content");
+
+        Path path = Path.of(tempDir.toString() + "/FileAdapterTest.save.txt");
+        File file = new File(path.toUri());
+
+        FileAdapter fileAdapter = new FileAdapter(file);
```
Comment (termitehuegel) [inline src/test/java/hwr/oop/group4/todo/persistence/FileAdapterTest.java:26]: Try to work with the Interface.

**Review by termitehuegel — CHANGES_REQUESTED** [review]
I have found some minor mistakes and have added a few recommendations in other places.

File: src/main/java/hwr/oop/group4/todo/persistence/Persisted.java [hunk 3]
```diff
@@ -0,0 +1,5 @@
+package hwr.oop.group4.todo.persistence;
+
+public interface Persisted {
+    String toString();
```
[the remark this answers is not in the document]
Comment (yxyx-github) [reply src/main/java/hwr/oop/group4/todo/persistence/Persisted.java]: What do you think about ```exportAsString```?

File: pom.xml [hunk 4]
```diff
@@ -69,6 +73,23 @@
             <version>${assertj.version}</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-core</artifactId>
+            <version>${mockito.version}</version>
```
[the remark this answers is not in the document]
Comment (yxyx-github) [reply pom.xml:79]: Why should I use it for the new dependencies while we don't use it for the rest? (Also I'm not familar with this.)

**Review by termitehuegel — APPROVED** [review]

**Pull request merged**